### PR TITLE
Fix datetime formatting v2

### PR DESCRIPTION
**Pull request by gbrian**
New PR without any extra commits. Sorry for #283

### CODE REVIEW

File: caravel/models.py [hunk 1]
```diff
@@ -577,7 +577,19 @@ def query(  # sqla
                 select_exprs += [timestamp_grain]
                 groupby_exprs += [timestamp_grain]
 
-            tf = '%Y-%m-%d %H:%M:%S.%f'
+            # UGLY: I guess correct way is to delegate on SQLAlchemy dialect
```
Comment (mistercrunch) [inline caravel/models.py]: Yeah let's figure out how to delegate it to SQLAlchemy, it knows how to do that

File: caravel/models.py [hunk 2]
```diff
@@ -577,7 +577,19 @@ def query(  # sqla
...
+            # UPDATE: Datetime depends on each dialect and I haven't found an easy way to manage
+            #         Maybe we can allow user to define its custome format at Database definition
+            def get_dtformat(type):
```
Comment (mistercrunch) [inline caravel/models.py]: FYI, you should never overwrite system words like `type`, the convention is to suffix with underscore, or just using something more specific.

**Comment by gbrian** [comment]
Ops, didn't see
image

File: caravel/models.py (hunk 1 above)
Comment (gyzmau) [inline caravel/models.py]: I agree, I got the issue yesterday. I figured out as well it was SQLAlchemy.
And I dont think Its "formating" issue, it's conversion issue. It's converting the datetime in Datetime2 (precision to the nanosecond).
I will try to look into it later.
For now I changed my columns in Datetime2 as a workaround for testing purpose.

**Pull request closed without merging**